weld/ast.rs

```diff
@@ -75,6 +75,181 @@ impl fmt::Display for ScalarKind {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum BuilderImplementationKind {
+    Local,
+    Global
+}
+
+impl fmt::Display for BuilderImplementationKind {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use ast::BuilderImplementationKind::*;
+        let text = match *self {
+            Local => "local",
+            Global => "global",
+        };
+        f.write_str(text)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Annotations {
+    builder_implementation: Option<BuilderImplementationKind>,
+
+    predicate: Option<bool>,
+    vectorize: Option<bool>,
+    tile_size: Option<i32>,
+    grain_size: Option<i32>,
+
+    size: Option<i32>,
+    branch_selectivity: Option<i32>, // Fractions of 10,000.
```

Comment:
Ah I see...and `Type` is used as a `HashMap` key in `llvm.rs`. I think eventually it might be good to just get rid of those traits for types but then we'd need a different way of storing a type in a map. This is okay for now.